### PR TITLE
Added structs for IAM import response

### PR DESCRIPTION
**Pull request by shtripat**

### CODE REVIEW

File: iam-migrate.go [hunk 1]
```diff
@@ -46,10 +52,10 @@ func (adm *AdminClient) ExportIAM(ctx context.Context) (io.ReadCloser, error) {
 }
 
 // ImportIAM makes an admin call to setup IAM  from imported content
-func (adm *AdminClient) ImportIAM(ctx context.Context, contentReader io.ReadCloser) error {
+func (adm *AdminClient) ImportIAM(ctx context.Context, contentReader io.ReadCloser) (sam SkippedIAMEntities, err error) {
```
Comment (harshavardhana) [inline iam-migrate.go:55]:
```suggestion
func (adm *AdminClient) ImportIAM(ctx context.Context, contentReader io.ReadCloser) (sam SkippedIAMEntities, err error) {
```

Never break existing APIs, introduce new APIs.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: iam-migrate.go (hunk 1 above)
Comment (harshavardhana) [reply iam-migrate.go:55]: ImportIAMV2

Comment (shtripat) [reply iam-migrate.go:55]: I am doing this second time, that's a sin ;) Will correct and will keep note of this.

**Review by harshavardhana — APPROVED** [review]

File: iam-migrate.go [hunk 2]
```diff
@@ -21,10 +21,16 @@ package madmin
 
 import (
 	"context"
+	"encoding/json"
 	"io"
 	"net/http"
 )
 
+type SkippedIAMEntities struct {
```
[the remark this answers is not in the document]
Comment (shtripat) [reply iam-migrate.go:29]: Ok, makes sense. Will change accordingly.

File: iam-migrate.go [hunk 3]
```diff
@@ -26,9 +26,11 @@ import (
 	"net/http"
 )
 
-type SkippedIAMEntities struct {
-	SkippedAccessKeys []string `json:"skippedAccessKeys,omitempty"`
-	SkippedDN         []string `json:"skippedDN,omitempty"`
+type ImportIAMResult struct {
+	Skipped []string `json:"skipped,omitempty"`
```
Comment (harshavardhana) [inline iam-migrate.go:30]: Is it going to be string alone? Isn't IAM various types of assets?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: iam-migrate.go (hunk 3 above)
Comment (shtripat) [reply iam-migrate.go:30]: accesskey and username are strings so kept like this

File: iam-migrate.go [hunk 4]
```diff
@@ -21,10 +21,48 @@ package madmin
...
+	Policies        []ErrEntity    `json:"policies,omitempty"`
+	Users           []ErrEntity    `json:"users,omitempty"`
+	Groups          []ErrEntity    `json:"groups,omitempty"`
+	ServiceAccounts []ErrEntity    `json:"serviceAccounts,omitempty"`
+	UserPolicies    []ErrPolEntity `json:"userPolicies,omitempty"`
+	GroupPolicies   []ErrPolEntity `json:"groupPolicies,omitempty"`
+	STSPolicies     []ErrPolEntity `json:"stsPolicies,omitempty"`
+}
+
+type ErrEntity struct {
```
Comment (harshavardhana) [inline iam-migrate.go:56]: we need to document each struct @shtripat

Comment (harshavardhana) [inline iam-migrate.go:56]:
```suggestion
type ImportIAMResult struct {
	Skipped IAMEntities    `json:"skipped,omitempty"`
	Removed IAMEntities    `json:"removed,omitempty"`
	Added   IAMEntities    `json:"added,omitmepty"`
	Failed  IAMErrEntities `json:"failed,omitmpty"`
}

type IAMEntities struct {
	Policies        []string              `json:"policies,omitmepty"`
	Users           []string              `json:"users,omitmepty"`
	Groups          []string              `json:"groups,omitempty"`
	ServiceAccounts []string              `json:"serviceAccounts,omitempty"`
	UserPolicies    []map[string][]string `json:"userPolicies,omitempty"`
	GroupPolicies   []map[string][]string `json:"groupPolicies,omitempty"`
	STSPolicies     []map[string][]string `json:"stsPolicies,omitempty"`
}

type IAMErrEntities struct {
	Policies        []ErrEntity    `json:"policies,omitempty"`
	Users           []ErrEntity    `json:"users,omitempty"`
	Groups          []ErrEntity    `json:"groups,omitempty"`
	ServiceAccounts []ErrEntity    `json:"serviceAccounts,omitempty"`
	UserPolicies    []ErrPolEntity `json:"userPolicies,omitempty"`
	GroupPolicies   []ErrPolEntity `json:"groupPolicies,omitempty"`
	STSPolicies     []ErrPolEntity `json:"stsPolicies,omitempty"`
}

type IAMErrEntity struct {
```

File: iam-migrate.go [hunk 5]
```diff
@@ -21,10 +21,48 @@ package madmin
...
+	Name  string `json:"name,omitempty"`
+	Error error  `json:"error,omitempty"`
+}
+
+type ErrPolEntity struct {
```
Comment (harshavardhana) [inline iam-migrate.go:61]:
```suggestion
type IAMErrPolicyEntity struct {
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**